csrc/fusion_segmenter.cpp

```diff
@@ -3938,7 +3931,7 @@ bool SegmentCandidateFinder::codeGenSupportedMerge(
 SchedulerType SegmentCandidateFinder::deriveSchedulerType(
     SegmentedGroup* group) {
   FUSER_PERF_SCOPE("SegmentCandidateFinder::deriveSchedulerType");
-  if (options_.only_segment_resharding_exprs) {
+  if (options_.custom_should_merge_groups != nullptr) {
```

Comment:
Why is this always `None`?

Reply:
It is nullptr by default, and if it is, we fallback to the traditional single device segmenter using the schedulers
does it answer your question?

Reply:
Sorry I misunderstood your question. I guess this one is more for @wujingyue -- here I'm only reproducing the previous behavior, but replacing the option "only_segment_resharding_exprs" with a more agnostic one.

The idea of returning None here has something to do with how FusionExecutorCache decide to lower segments. However, this is not used in MultiDeviceExecutor

Reply:
Yes, agreed. For the record, this hack has been present for quite a long time now. Let me add the comment as you suggest

Reply:
> The overall design of the segmenter is tightly coupled with scheduling, so it is assumed to have this scheduler type

That's correct, @naoyam. FWIW, this flag is only turned on for MultiDeviceExecutor. In FusionExecutorCache, schedulers test isResharding as you suggested.